lib/puppet/util.rb

```diff
@@ -479,7 +479,7 @@ def safe_posix_fork(stdin = $stdin, stdout = $stdout, stderr = $stderr, &block)
 
       begin
         Dir.foreach('/proc/self/fd') do |f|
-          if f != '.' && f != '..' && f.to_i >= 3
+          if %{^\d+$}.match?(f) && f.to_i >= 3
```

Comment:
If you want this to work, it needs to be
```suggestion
          if %r{^\d+$}.match?(f) && f.to_i >= 3
```